ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/LgrDB.hs

```diff
@@ -239,22 +253,25 @@ initFromDisk
   => LgrDbArgs Identity m blk
   -> Tracer m (TraceReplayEvent blk ())
   -> ImmutableDB m blk
-  -> m (LedgerDB' blk, Word64)
-initFromDisk LgrDbArgs { lgrHasFS = hasFS, .. }
-             replayTracer
-             immutableDB = wrapFailure (Proxy @blk) $ do
+  -> m (LedgerDB' blk, Word64, OnDiskLedgerStDb m (ExtLedgerState blk))
+initFromDisk args replayTracer immutableDB = wrapFailure (Proxy @blk) $ do
+    onDiskLedgerStDb <- LedgerDB.mkOnDiskLedgerStDb lgrHasFSLedgerSt
+    -- TODO: is it correct that we pick a instance of the 'OnDiskLedgerStDb' here?
```

Comment:
There might be a backend (like the first one we'll use) that don't require a file  handle.